src/stitching/errors.ts

```diff
@@ -0,0 +1,91 @@
+import { GraphQLResolveInfo, responsePathAsArray } from 'graphql';
+import { locatedError } from 'graphql/error';
+
+const ERROR_SYMBOL = Symbol('Schema Merging Error');
+
+export function annotateWithChildrenErrors(
+  object: any,
+  childrenErrors: Array<{ path?: Array<string | number> }>,
+): any {
+  if (childrenErrors && childrenErrors.length > 0) {
+    if (Array.isArray(object)) {
+      const byIndex = {};
+      childrenErrors.forEach(error => {
+        const index = error.path[1];
+        const current = byIndex[index] || [];
+        current.push({
+          ...error,
+          path: error.path.slice(1),
+        });
+        byIndex[index] = current;
+      });
+      return object.map((item, index) =>
+        annotateWithChildrenErrors(item, byIndex[index]),
+      );
+    } else {
+      return {
+        ...object,
+        [ERROR_SYMBOL]: childrenErrors.map(error => ({
+          ...error,
+          path: error.path.slice(1),
+        })),
+      };
+    }
+  } else {
+    return object;
+  }
+}
+
+export function getErrorsFromParent(
+  object: any,
+  fieldName: string,
+): {
+  ownError?: any;
+  childrenErrors?: Array<{ path?: Array<string | number> }>;
+} {
+  const errors = (object && object[ERROR_SYMBOL]) || [];
+  const childrenErrors: Array<{ path?: Array<string | number> }> = [];
+  for (const error of errors) {
+    if (error.path.length === 1 && error.path[0] === fieldName) {
+      return {
+        ownError: error,
+      };
+    } else if (error.path[0] === fieldName) {
+      childrenErrors.push(error);
+    }
+  }
+  return {
+    childrenErrors,
```

Comment:
I don't think I fully understand the logic, but is there ever a situation where you'd want to return both `ownError` and `childrenErrors`?

Reply:
No. I guess I should use algebraic type with key.

Reply:
Done.